### PR TITLE
Updating archiver to have more of the functions declared

### PR DESCRIPTION
**Pull request by Dirrk**
DOCS from here https://archiverjs.com/docs/Archiver.html

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by mhegazy** [comment]
@Dirrk CI still failing, can you please address these failures.

File: types/archiver/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for archiver 1.3
+// Type definitions for archiver v1.3.0
```
Comment (andy-ms) [inline types/archiver/index.d.ts]: We actually prefer just `MAJOR.MINOR`. Undo this change.

File: types/archiver/index.d.ts [hunk 2]
```diff
@@ -12,32 +12,84 @@
...
 
+    interface EntryData extends NameInterface {
+        date?: string | any;
+        mode?: number;
+        prefix?: string;
+        stats?: FS.Stats;
+    }
+
+    interface ZipEntryData extends EntryData {
+   store?: boolean;
```
Comment (andy-ms) [inline types/archiver/index.d.ts]: Fix indent

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has failing CI tests and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.
